syntax = "proto3";

package cpp2sky;

enum Protocol {
  GRPC = 0;
  REST = 1;
}

message TracerConfig {
  // Tracer protocol.
  Protocol protocol = 1;

  // Service name
  string service_name = 2;

  // Instance name
  string instance_name = 3;

  // OAP address.
  string address = 4;

  // OAP token.
  string token = 5;

  // The size of buffer it stores pending messages.
  uint32 delayed_buffer_size = 6;

  // If the operation name of the first span is included in this set, 
  // this segment should be ignored.
  repeated string ignore_operation_name_suffix = 8;

  // CDS sync request interval. If this value is zero, CDS feature will be disabled.
  uint32 cds_request_interval = 7;
}